fleet/fleet.go

```diff
@@ -131,6 +132,15 @@ type Fleet interface {
 func NewFleet(config Config) (Fleet, error) {
 	var trans http.RoundTripper
 
+	tunneling := config.Tunnel.Tunnel != ""
+	// If a tunnel in provided we need to overwrite the http.Transport.Dial function
+	// to use the tunnel
+	if tunneling {
+		trans = &http.Transport{
+			Dial: config.Tunnel.NewDialFunc(config),
+		}
+	}
```

Comment:
This looks like the transport is going to be overwritten in the upcoming switch, in case `config.Endpoint.Scheme` is `unix` or `file`.